frontend/ui/data/onetime_migration.lua

```diff
@@ -585,5 +586,22 @@ if last_migration_date < 20230710 then
     end
 end
 
+-- 20230727, Statistics plugin null id_book in page_stat_data
+if last_migration_date < 20230727 then
+    logger.info("Performing one-time migration for 20230727")
+    local conn = SQ3.open(DataStorage:getSettingsDir() .. "/statistics.sqlite3")
+    local ok, value = pcall(conn.exec, conn, "PRAGMA table_info('page_stat_data')")
```

Comment:
Is that OK if the statistics plugin is disabled (and the user has deleted the db) ?

Reply:
>plugin is disabled

Still performs the deletion, should be fine and intended behavior.

>the user has deleted the db

A warning will be shown in the log, seems an empty db will be created due to `SQ3.open()`. Should not be too big a deal though?

Reply:
You could `pcall(SQ3.open, ...)` just for consistency ?
(I guess it's more likely to happen than your `db not compatible` check :) )

Reply:
"Could not open blah" warnings are the norm for migration blocks, FWIW. We only call stuff that would otherwise crash ;).

Reply:
Added check for existence of file.